Extension/src/Debugger/configurationProvider.ts

```diff
@@ -526,6 +519,15 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
         }
     }
 
+    public async getLaunchConfigs(folder: vscode.WorkspaceFolder | undefined): Promise<vscode.WorkspaceConfiguration[] | undefined> {
+        let configs: vscode.WorkspaceConfiguration[] | undefined = vscode.workspace.getConfiguration('launch', folder).get('configurations');
+        if (!configs) {
+            return undefined;
+        }
+        configs = configs.filter(config => (config.name && config.type === DebuggerType.cppvsdbg || config.type === DebuggerType.cppdbg) && config.request === "launch");
```

Comment:
```
 (config.name && config.type === DebuggerType.cppvsdbg || config.type === DebuggerType.cppdbg) && config.request === "launch"
```

The parenthesis `( )` may need clean up. Is this filter to check for:

- config name exits AND
- config request is launch AND
- `(`config type is either cppvsdg OR cppdbg`)`  ?

Reply:
right, I cleaned it up